File: 02-programacion-basica/11-archivos-y-persistencia-basica/scala/soluciones/RetoTareasArchivo.scala
import java.nio.file.{Files, Path}
import java.nio.charset.StandardCharsets
import scala.jdk.CollectionConverters.*

@main def retoTareasArchivo(): Unit =
  val archivo = Path.of("tareas-devpath.txt")
  val tareas = List("Estudiar Java", "Practicar Scala", "Revisar Git")

  // asJava adapta la colección Scala a la API de java.nio.
  Files.write(archivo, tareas.asJava, StandardCharsets.UTF_8)
  val guardadas = Files.readAllLines(archivo, StandardCharsets.UTF_8).asScala
  guardadas.zipWithIndex.foreach { case (tarea, i) => println(s"${i + 1}. $tarea") }
  Files.deleteIfExists(archivo)

File: 02-programacion-basica/11-archivos-y-persistencia-basica/java/soluciones/RetoTareasArchivo.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;

public class RetoTareasArchivo {
    public static void main(String[] args) {
        Path archivo = Path.of("tareas-devpath.txt");
        List<String> tareas = List.of("Estudiar Java", "Practicar Scala", "Revisar Git");

        try {
// Files.write() persiste cada elemento de la lista como una línea del archivo.
            Files.write(archivo, tareas);
            List<String> guardadas = Files.readAllLines(archivo);
            for (int i = 0; i < guardadas.size(); i++) {
                System.out.printf("%d. %s%n", i + 1, guardadas.get(i));
            }
            Files.deleteIfExists(archivo);
        } catch (IOException e) {
            System.out.println("Error de E/S: " + e.getMessage());
        }
    }
}
